tests/ipsec/ipsec3hosts.pm

```diff
@@ -0,0 +1,126 @@
+# SUSE's openQA tests
+#
+# Copyright 2017-2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary:  Basic preparation before any IPSec test
+# Maintainer: Kernel QE <kernel-qa@suse.de>
+
+use Mojo::Base "ipsecbase", -signatures;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+use lockapi;
+use network_utils;
+
+
+sub run ($self) {
+
+    select_serial_terminal;
+
+    $self->prepare_env();
+
+    # This case using following 3 hosts scenario
+    # host(left_ip) -------(middle_ip_01) router(middle_ip_01) ------ host(right_ip)
+
+    my $setup = {
+        'left_ip' => "2001:1:1:1::2",
+        'right_ip' => "2002:1:1:1::2",
+        'left_net' => "2001:1:1:1::/64",
+        'right_net' => "2002:1:1:1::/64",
+        'middle_ip_01' => "2001:1:1:1::1",
+        'middle_ip_02' => "2002:1:1:1::1",
+        'middle_net_01' => "2001:1:1:1::/64",
+        'middle_net_02' => "2002:1:1:1::/64",
+    };
+
+    my $ipsec_setting_left = {
+        'local_ip' => $setup->{left_ip},
+        'remote_ip' => $setup->{right_ip},
+        'new_local_net' => $setup->{left_net},
+        'new_remote_net' => $setup->{right_net}
+    };
+
+    my $ipsec_setting_right = {
+        'local_ip' => $setup->{right_ip},
+        'remote_ip' => $setup->{left_ip},
+        'new_local_net' => $setup->{right_net},
+        'new_remote_net' => $setup->{left_net}
+    };
+
+    record_info('IPSEC_SETUP', get_var('IPSEC_SETUP'));
+
+    if (get_var('IPSEC_SETUP') eq 'left') {
+        $self->add_ipv6_addr(ip => $setup->{left_ip}, plen => $self->get_net_prefix_len($setup->{left_net}));
+        $self->check_ipv6_addr();
+        barrier_wait 'IPSEC_IP_SETUP_DONE';
+        script_run("ping -c 1 $setup->{middle_ip_01}");
+        script_run("ip neighbor  show");
+        $self->add_ipv6_route(dst => $setup->{right_ip}, via => $setup->{middle_ip_01});
+        barrier_wait 'IPSEC_ROUTE_SETUP_DONE';
+        script_run("ip a");
+        script_run("ip -6 route");
+        script_run("ping -c 1 $setup->{middle_ip_01}");
+        script_run("ping -c 1 $setup->{right_ip}");
```

Comment:
I would personally use shell script with `/bin/bash -eux` (more readable code, but it would still print commands and quit on first error), but sure, it can stay like this.

Reply:
current approach is correct one, but i would use `assert_script_run`  to properly fail

Reply:
got it. will do s/script_run/assert_script_run/gc